templates/haproxy.systemd.j2

```diff
@@ -14,6 +15,23 @@ ExecReload=/usr/sbin/haproxy -f $CONFIG -c -q $EXTRAOPTS
 ExecReload=/bin/kill -USR2 $MAINPID
 KillMode=mixed
 Restart=always
+{% else %}
+LimitNOFILE=262144
+EnvironmentFile=-/etc/default/haproxy
+{% if haproxy_version is version('2.0', '>') %}
+Environment="CONFIG=/etc/haproxy/haproxy.cfg" "PIDFILE=/run/haproxy.pid" "EXTRAOPTS=-S /run/haproxy-master.sock"
```

Comment:
There are two definition of `Environment` depending of the logic but it is defined three time. The check on version 2.0 or greater should stand on its own.

Also I do you need to define /run/haproxy-master.sock here ?